bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/CipherCallback.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.mongodb.crypt.capi;
+
+import com.mongodb.crypt.capi.CAPI.mongocrypt_crypto_fn;
+import com.sun.jna.Pointer;
+
+import javax.crypto.Cipher;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import java.security.GeneralSecurityException;
+
+import static com.mongodb.crypt.capi.CAPI.MONGOCRYPT_STATUS_ERROR_CLIENT;
+import static com.mongodb.crypt.capi.CAPI.mongocrypt_status_set;
+import static com.mongodb.crypt.capi.CAPIHelper.toByteArray;
+import static com.mongodb.crypt.capi.CAPIHelper.writeByteArrayToBinary;
+
+class CipherCallback implements mongocrypt_crypto_fn {
+    private final String algorithm;
+    private final String transformation;
+    private final int mode;
+
+    CipherCallback(final String algorithm, final String transformation, final int mode) {
+        this.algorithm = algorithm;
+        this.transformation = transformation;
+        this.mode = mode;
+    }
+
+    @Override
+    public boolean crypt(final Pointer ctx, final CAPI.mongocrypt_binary_t key, final CAPI.mongocrypt_binary_t iv,
```

Comment:
Any reason its `CAPI.mongocrypt_binary_t` and not just `monogcrypt_binary_t`?  Elsewhere in the lib the shorter form seems to be preferred

Reply:
Ah, it happened when I did an auto-refactoring to move the callbacks out of MongoCryptImpl.java.  Fixed it in all of the callbacks.